src/common/hostname.cc

```diff
@@ -14,12 +14,14 @@
 
 #include "common/hostname.h"
 
-#include <unistd.h>
-
 std::string ceph_get_hostname()
 {
-  char buf[1024];
-  gethostname(buf, 1024);
+#ifndef HOST_NAME_MAX
+#define HOST_NAME_MAX 1024
+#endif
+  char buf[HOST_NAME_MAX + 1];
+  assert(gethostname(buf, sizeof(buf)) == 0);
```

Comment:
please avoid using the side effect of `assert()`, it could be optimized out in future if `NDEBUG=1`.

Reply:
@shinobu-x this is not addressed